crates/adkg-cli/src/onlyswaps.rs

```diff
@@ -0,0 +1,128 @@
+use crate::keygen::PrivateKeyMaterial;
+use crate::{AdkgPublic, AdkgSecret, GroupConfig};
+use alloy::hex;
+use alloy::primitives::FixedBytes;
+use ark_bn254::G2Affine;
+use base64::Engine;
+use base64::prelude::BASE64_STANDARD;
+use clap::Parser;
+use config::agent::AgentConfig;
+use config::app::{AppConfig, Libp2pConfig};
+use config::keys::{Bn254SecretKey, Libp2pKeyWrapper};
+use config::network::NetworkConfig;
+use config::signing::{CommitteeConfig, MemberConfig};
+use libp2p::Multiaddr;
+use std::fs;
+use std::net::Ipv4Addr;
+use std::num::NonZeroU16;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::time::Duration;
+use url::Url;
+use utils::serialize::point::PointDeserializeCompressed;
+
+#[derive(Parser, Debug)]
+pub struct GenerateOnlyswapsConfig {
+    #[arg(
+        long = "private",
+        help = "the private key file from the generate-keys step"
+    )]
+    pub operator_private: PathBuf,
+
+    #[arg(long = "group", help = "the group file used to run the DKG")]
+    pub group: PathBuf,
+
+    #[arg(long = "adkg-public", help = "the ADKG public output file")]
+    pub adkg_public: PathBuf,
+
+    #[arg(long = "adkg-private", help = "the ADKG private output file")]
+    pub adkg_private: PathBuf,
+
+    #[arg(
+        long = "multiaddr",
+        help = "the multiaddr your node is running at to peer with other nodes"
+    )]
+    pub multiaddr: Multiaddr,
+
+    #[arg(
+        long = "member-id",
+        help = "the index of your node in the final committee"
+    )]
+    pub member_id: NonZeroU16,
+
+    #[arg(long, help = "the address of the router contract on each chain")]
+    pub router_address: Option<FixedBytes<20>>,
+}
+
+pub(crate) async fn generate_onlyswaps_config(args: GenerateOnlyswapsConfig) -> anyhow::Result<()> {
+    // load from all the relevant files
+    let secret_key: PrivateKeyMaterial =
+        toml::from_str(&fs::read_to_string(&args.operator_private)?)?;
+    let group: GroupConfig = toml::from_str(&fs::read_to_string(&args.group)?)?;
+    let shared_pub: AdkgPublic = toml::from_str(&fs::read_to_string(&args.adkg_public)?)?;
+    let shared_priv: AdkgSecret = toml::from_str(&fs::read_to_string(&args.adkg_private)?)?;
```

Comment:
I'd consider adding context to the errors with something like 
```
toml::from_str(
    &fs::read_to_string(&args.adkg_private).context("failed to read adkg private key")?
)
.context("failed to parse adkg private key")?;
```

to prevent having a generic "file not found", or "Error: TOML parse error at line 1, column 1"